src/model.hpp

```diff
@@ -220,16 +213,18 @@ class model {
 
     // only thread safe if called outside the run() method
     void add_artificial_spike(cell_member_type source, time_type tspike) {
-        current_spikes().get().push_back({source, tspike});
+        if (domain_.is_local_gid(source.gid)) {
```

Comment:
Should we add a TODO to remove this function when we have 'real' artificial cells. To prevent that we have non-threadsave code laying about?